### PR TITLE
feat(text-input): add sr automatic announcement on error message

### PR DESCRIPTION
**Pull request by nataliadelmar**
# Description

As we can see in the a11y audit reports, if an error gets displayed related to an input, this should be announced automatically by SR. 

# Links

[*Links to relevent resources.*](https://confluence-eng-gpk2.cisco.com/conf/display/WTWC/2.+INPUTS#:~:text=%E2%9A%AA%20ALLOW%20SR%20TO%20ANNOUNCE%20IF%20A%20TEXT%20INPUT%20HAS%20AN%20ERROR)

### CODE REVIEW

File: src/components/TextInput/TextInput.unit.test.tsx [hunk 1]
```diff
@@ -14,9 +15,21 @@ jest.mock('uuid', () => {
 });
 
 describe('<TextInput/>', () => {
+  let announceSpy;
+  let container;
```
Comment (dominiccarrington) [inline src/components/TextInput/TextInput.unit.test.tsx:19]: nitpick: these can be typed as
```ts
let announceSpy: jest.SpiedFunction<(typeof screenReaderAnnouncer)['default']['announce']>;
let container: ReactWrapper;
```

File: src/components/TextInput/TextInput.unit.test.tsx [hunk 2]
```diff
@@ -299,5 +308,22 @@ describe('<TextInput/>', () => {
...
+
+      container = await mountAndWait(
+        <TextInput
+          clearAriaLabel="Clear this input"
+          label="text-input"
+          messageArr={[message1, message2]}
+        />
+      );
+
+      expect(announceSpy).toHaveBeenCalledWith({ body: 'test1' }, 'test-ID');
```
Comment (dominiccarrington) [inline src/components/TextInput/TextInput.unit.test.tsx:326]: question: should it announce  all of the errors?

File: src/components/ComboBox/ComboBox.unit.test.tsx
Comment (dominiccarrington) [inline src/components/ComboBox/ComboBox.unit.test.tsx:1]: Thanks for sorting out the hierarchy of these tests!

**Review by dominiccarrington — APPROVED** [review]
LGTM, just a nitpick and a question.

File: src/components/TextInput/TextInput.unit.test.tsx (hunk 2 above)
Comment (nataliadelmar) [reply src/components/TextInput/TextInput.unit.test.tsx:326]: this textinput implementation is assuming that there will only be 1 error in the messageArr, and we should only sr announce errors, not any other message. So im leveraging on that

**Pull request merged**